core/services/directrequestocr/drlistener.go

```diff
@@ -204,6 +207,14 @@ func ExtractRawBytes(input []byte) ([]byte, error) {
 	return utils.TryParseHex(string(input))
 }
 
+func (l *DRListener) getNewHandlerContext() (context.Context, context.CancelFunc) {
+	timeoutSec := l.pluginConfig.ListenerEventHandlerTimeoutSec
+	if timeoutSec == 0 {
+		return context.WithCancel(l.serviceContext)
```

Comment:
I noticed we have both `chStop` and `serviceContext` that are bound to service lifetime.
A suggested Core pattern is to use just `serviceContext`. You could safely remove `chStop` and replace with `serviceContext`.

Reply:
Will do separately, thanks!